### PR TITLE
Add support for context in nback.llvm

### LINKED ISSUE #114: Add support for context in nback.llvm
@jclark for this I am thinking of adding fallowing to the public API

1. Public record type `Context` to represent [LLVMContextRef](https://llvm.org/doxygen/group__LLVMCSupportTypes.html#ga9c43e01525516ff6b4feab5166c5b1da). Initially this can be an empty place holder since we only need this to have API similar to what we get with #19.
2. Public function `createContext() returns Context` similar to https://llvm.org/doxygen/group__LLVMCCoreContext.html#gaac4f39a2d0b9735e64ac7681ab543b4c.
3. All public constructors ([`Module`](https://llvm.org/doxygen/group__LLVMCCoreModule.html#ga6bf4a0a387cf7fe04cbe0438cdb36a51), [`Builder`](https://llvm.org/doxygen/group__LLVMCCoreInstructionBuilder.html#ga720a9461e32f7c3d72011bfd6724f92f)) will take `Context` as their first argument(Java api needs a name for module as well so I'll modify our constructor to take an optional name parameter)
4. Other constructors such as those for functions will get the context from the parent (module in the case of function)

I am assuming when it comes for types we can ignore the context even though we use it in other places (ex for `i64` use [`LLVMInt64Type`](https://llvm.org/doxygen/group__LLVMCCoreTypeInt.html#ga0a46c5d828749b932cc728159fcf129a) instead of [`LLVMInt64TypeInContext`](https://llvm.org/doxygen/group__LLVMCCoreTypeInt.html#ga213ec0fe49543773320798d7cb619746) in the C api. If not constant constructors (ex:`intConst`) will also need `Context` as an argument. (In otherplaces we should be able to get the context from parent (In most cases builder)).

### PR DESCRIPTION
**Pull request by heshanpadmasiri**
## Purpose
Add minimal support for context such that we can get consistent API when we implement #19 we get the same API
Resolves #114 

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

**Review by jclark — CHANGES_REQUESTED** [review]
See https://github.com/ballerina-platform/nballerina/issues/114#issuecomment-865671414

File: compiler/modules/nback.llvm/llvm.bal [hunk 1]
```diff
@@ -653,3 +658,15 @@ function omitSpaceBefore(string word) returns boolean {
...
+// Corresponds to LLVMContextRef
+public type Context readonly & record {||};
+
+// Corresponds to LLVMContextCreate
+public function contextCreate() returns Context {
+    return {};
+}
+
+// Corresponds to LLVMContextDispose
+public function contextDispose(Context context) {
```
Comment (jclark) [inline compiler/modules/nback.llvm/llvm.bal]: Please remove. This doesn't work with a GC-language.

**Review by jclark — CHANGES_REQUESTED** [review]

File: compiler/modules/nback.llvm/llvm.bal [hunk 2]
```diff
@@ -85,6 +85,24 @@ public function constNull(PointerType ty) returns PointerValue {
...
 }
 
+// Corresponds to LLVMContextRef
+public class Context {
+    // Corresponds to LLVMContextCreate
+    public function init() {
+        
+    }
+
+    public function createModuleInContext() returns Module {
```
Comment (jclark) [inline compiler/modules/nback.llvm/llvm.bal]: Just `createModule`.

File: compiler/modules/nback.llvm/llvm.bal [hunk 3]
```diff
@@ -85,6 +85,24 @@ public function constNull(PointerType ty) returns PointerValue {
...
+        return new(self);
+    }
+
+    public function  createBuilderInContext() returns Builder {
```
Comment (jclark) [inline compiler/modules/nback.llvm/llvm.bal]: Just `createBuilder`

File: compiler/modules/nback.llvm/llvm.bal [hunk 4]
```diff
@@ -85,6 +85,24 @@ public function constNull(PointerType ty) returns PointerValue {
...
+        return new(self);
+    }
+
+    // Corresponds to LLVMContextDispose
+    public function dispose(){}
```
Comment (jclark) [inline compiler/modules/nback.llvm/llvm.bal]: Remove. This doesn't work with a GC language like Ballerina.

**Review by jclark — CHANGES_REQUESTED** [review]

File: compiler/modules/nback.llvm/llvm.bal (hunk 4 above)
Comment (heshanpadmasiri) [reply compiler/modules/nback.llvm/llvm.bal]: Fixed

File: compiler/modules/nback.llvm/llvm.bal (hunk 3 above)
Comment (heshanpadmasiri) [reply compiler/modules/nback.llvm/llvm.bal]: Fixed

File: compiler/modules/nback.llvm/llvm.bal (hunk 2 above)
Comment (heshanpadmasiri) [reply compiler/modules/nback.llvm/llvm.bal]: Fixed

File: compiler/modules/nback.llvm/llvm.bal (hunk 1 above)
Comment (heshanpadmasiri) [reply compiler/modules/nback.llvm/llvm.bal]: Fixed

**Comment by jclark** [comment]
This broke the build because the `trunc` test wasn't fixed.

**Pull request merged**